File: app/src/main/java/com/aserbao/aserbaosandroid/comon/base/viewHolder/ClassViewHolder.java
package com.aserbao.aserbaosandroid.comon.base.viewHolder;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.aserbao.aserbaosandroid.AUtils.utils.ConstantUtils;
import com.aserbao.aserbaosandroid.R;
import com.aserbao.aserbaosandroid.comon.base.beans.BaseRecyclerBean;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ClassViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.item_tv)
        public TextView mItemTv;
        @BindView(R.id.item_card_view)
        public CardView mItemCardView;
        public ClassViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }


        public void setDataSource(Activity mActivity, BaseRecyclerBean classBean){
            mItemCardView.setBackgroundResource(ConstantUtils.getDrawable());
            mItemTv.setText(classBean.getName());
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent intent = new Intent(mActivity, classBean.getClazz());
                    mActivity.startActivity(intent);
                }
            });
        }
    }

File: base/src/main/java/com/example/base/base/viewHolder/ClassViewHolder.kt
package com.example.base.base.viewHolder

import android.app.Activity
import android.content.Intent
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.example.base.base.beans.BaseRecyclerBean
import com.example.base.utils.data.ASourceUtil
import kotlinx.android.synthetic.main.base_recycler_view_class_item.view.*

class ClassViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView!!) {
    fun setDataSource(mActivity: Activity, classBean: BaseRecyclerBean) {
        itemView.item_card_view.setBackgroundResource(ASourceUtil.getDrawable())
        itemView.item_tv.text = classBean.name
        itemView.setOnClickListener {
            val intent = Intent(mActivity, classBean.clazz)
            mActivity.startActivity(intent)
        }
    }


}